design/one-pager-component-mutable-and-versioning.md

```diff
@@ -0,0 +1,475 @@
+      kind: ContainerizedWorkload
+      spec:
+        containers:
+        - name: my-cool-workload
+          image: example/very-cool-workload:0.1.2@sha256:verytrustworthyhash
+          cmd:
+          - "bash lscpu"
+```
+
+Make a change to `component.yaml`:
+
+```shell script
+           cmd:
+-            - "bash lscpu"
++            - "bash top"
+```
+
+A new `ControllerRevision` will be automatically generated:
+
+```shell script
+$ kubectl get controllerrevisions
+NAMESPACE     NAME                  CONTROLLER               REVISION   AGE
+default       frontend-c8bb659c5    core.oam.dev/component   1          2d15h
+default       frontend-a75588698    core.oam.dev/component   2          2d14h
+
+$ kubectl get controllerrevision frontend-a75588698 -o yaml
+apiVersion: apps/v1
+kind: ControllerRevision
+metadata:
+  labels:
+    controller-revision-hash: a75588698
+  name: frontend-a75588698
+  namespace: default
+  ownerReferences:
+  - apiVersion: core.oam.dev/v1alpha2
+    blockOwnerDeletion: true
+    controller: true
+    kind: Component
+    name: frontend
+revision: 2
+data:
+  spec:
+    workload:
+      apiVersion: core.oam.dev/v1alpha2
+      kind: ContainerizedWorkload
+      spec:
+        containers:
+        - name: my-cool-workload
+          image: example/very-cool-workload:0.1.2@sha256:verytrustworthyhash
+          cmd:
+          - "bash top"
+```
+
+Now that the basic controller revision are in place, we can use them in Trait and ApplicationConfiguration.
+
+### A real running workload instance will have the same name with revision
+
+If we create an `ApplicationConfiguration` with definition : 
+
+```yaml
+apiVersion: core.oam.dev/v1alpha2
+kind: ApplicationConfiguration
+metadata:
+  name: example-appconfig
+spec:
+  components:
+    - componentName: frontend
+```
+
+The real workload instance will be created with the same name as the controllerRevision.
+
+```yaml
+  apiVersion: core.oam.dev/v1alpha2
+  kind: ContainerizedWorkload
+  metadata:
+    name: frontend-a75588698
+  spec:
+    containers:
+    - name: my-cool-workload
+      image: example/very-cool-workload:0.1.2@sha256:verytrustworthyhash
+      cmd:
+      - "bash lscpu"
+```
+
+
+### ControllerRevision could be referenced by Trait 
+
+Some traits can control version of component which means they can specify a component revision and won't be affected
+by the change of component.
+
+For these traits, they have to declare `revision` field in TraitDefinition like below:
+
+```yaml
+apiVersion: core.oam.dev/v1alpha2
+kind: TraitDefinition
+metadata:
+  name: fancytraits.core.oam.dev
+spec:
+  revision:
+    alwaysControl: true
+    revisonPath:
+      - "spec.revision"
+  definitionRef:
+    name: fancytraits.core.oam.dev
+```
+
+#### Revision field rule
+
+Versioning mechanism will be related with `revision` field.
+
+- If no traits binding with a component have revision field, OAM runtime will always update workload directly.
```

Comment:
The essential difference is workload CR's name is revisioned or not. The narrative of "update new workload directly" is not accurate.

Also, for both cases, OAM runtime will generate `ControllerRevision` object to track the change of component.